parquet/src/file/serialized_reader.rs

```diff
@@ -568,6 +568,62 @@ impl<R: ChunkReader> SerializedPageReader<R> {
             physical_type: meta.column_type(),
         })
     }
+
+    /// Similar to `peek_next_page`, but returns the offset of the next page instead of the page metadata.
+    /// Unlike page metadata, an offset can uniquely identify a page.
+    ///
+    /// This is used when we need to read parquet with row-filter, and we don't want to decompress the page twice.
+    /// This function allows us to check if the next page is being cached or read previously.
+    pub fn peek_next_page_offset(&mut self) -> Result<Option<usize>> {
```

Comment:
I think it would be better to make this non pub as I don't think it needs to be a public API (`peek_next_page` is not `pub` either)

I removed the `pub` and got a "not used" error, but I think that is because it will be used in the future as part of https://github.com/apache/arrow-rs/pull/6921

This seemed to work for me locally:
```suggestion
    #[cfg(test)]
    fn peek_next_page_offset(&mut self) -> Result<Option<usize>> {
```